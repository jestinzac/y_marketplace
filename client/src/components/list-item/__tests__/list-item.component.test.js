import { render, screen } from "@testing-library/react";

import ListItem from "../list-item.component";

describe("ListItem", () => {
  const mockData = {
    id: 1,
    tier: "plumber", 
    headline: "My toilet won't flush", 
    description: "There is a problem with my toilet and I need someone to fix it.", 
    date_added: "2020-01-15T10:47:54Z", 
    area: "Brooklyn, NYC"
  };

  it("should render the respective list item", () => {
    Object.defineProperty(global.navigator, "language", {
      get: () => "en-GB",
      enumerable: true,
      configurable: true,
    });

    const { asFragment } = render(
      <ListItem {...mockData} />
    );

    expect(screen.getByText("My toilet won't flush")).toBeInTheDocument();
    expect(screen.getByTestId("calender-svg-icon")).toBeInTheDocument();
    expect(screen.getByTestId("map-pin-svg-icon")).toBeInTheDocument();
    expect(screen.getByTestId("formatted-date").textContent).toBe("15/01/2020, 10:47");
    expect(screen.getByTestId("list-item")).toHaveAttribute("id", "1");
    expect(asFragment()).toMatchSnapshot();
  });
});
