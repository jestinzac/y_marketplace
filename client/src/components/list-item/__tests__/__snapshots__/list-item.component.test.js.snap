// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem should render the respective list item 1`] = `
<DocumentFragment>
  <article
    class="w-auto p-6 m-5 bg-white rounded-md overflow-hidden shadow-lg hover:shadow-xl list-item"
    data-testid="list-item"
    id="1"
  >
    <span
      class="rounded-full bg-purple-200 py-1 px-4 text-xs font-bold text-purple-600 tag"
    >
      plumber
    </span>
    <h2
      class="text-3xl py-4 font-bold text-blue-800 description headline"
    >
      My toilet won't flush
    </h2>
    <p
      class="text-xl py-3 pb-20 font-medium text-gray-900"
    >
      There is a problem with my toilet and I need someone to fix it.
    </p>
    <div
      class="py-4 flex justify-between items-center bottom-info"
    >
      <div
        class="flex gap-2"
      >
        <svg
          class="w-6 h-6"
          data-testid="calender-svg-icon"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6.75 3v2.25M17.25 3v2.25M3 18.75V7.5a2.25 2.25 0 012.25-2.25h13.5A2.25 2.25 0 0121 7.5v11.25m-18 0A2.25 2.25 0 005.25 21h13.5A2.25 2.25 0 0021 18.75m-18 0v-7.5A2.25 2.25 0 015.25 9h13.5A2.25 2.25 0 0121 11.25v7.5"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
        <span
          class="date content-end"
          data-testid="formatted-date"
        >
          15/01/2020, 10:47
        </span>
      </div>
      <div
        class="flex gap-2"
      >
        <svg
          class="w-6 h-6"
          data-testid="map-pin-svg-icon"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15 10.5a3 3 0 11-6 0 3 3 0 016 0z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
          <path
            d="M19.5 10.5c0 7.142-7.5 11.25-7.5 11.25S4.5 17.642 4.5 10.5a7.5 7.5 0 1115 0z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
        <span
          class="location"
        >
          Brooklyn, NYC
        </span>
      </div>
    </div>
  </article>
</DocumentFragment>
`;
